packages/subiquity_client/lib/src/http_unix_client.dart

```diff
@@ -86,7 +94,10 @@ class HttpUnixClient extends BaseClient {
 
   @override
   void close() {
-    _socket.close();
+    if (_socket != null) {
+      _socket?.close();
```

Comment:
`Socket.close()` is asynchronous. Would await help?

Side note: is there a chance there might be buffered data? Should the socket be flushed before closing?